rust: clippy is not happy

**What:**

The rust workflow fails because of a linter error:

https://github.com/Orange-OpenSource/its-client/actions/runs/12767758905/job/35586775284?pr=259

```
$ cd rust
$ cargo clippy -- -D clippy::all
    [...]
    Checking libits-client v2.0.0 (/home/runner/work/its-client/its-client/rust)
error: this lifetime isn't used in the impl
  --> src/transport/mqtt/mqtt_client.rs:35:6
   |
35 | impl<'client> MqttClient {
   |      ^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#extra_unused_lifetimes
   = note: `-D clippy::extra-unused-lifetimes` implied by `-D clippy::all`
   = help: to override `-D clippy::all` add `#[allow(clippy::extra_unused_lifetimes)]`

error: could not compile `libits-client` (lib) due to 1 previous error
Error: Process completed with exit code 101.
```

--- a/rust/src/transport/mqtt/mqtt_client.rs
+++ b/rust/src/transport/mqtt/mqtt_client.rs
@@ -32,7 +32,7 @@ pub struct MqttClient {
     client: AsyncClient,
 }
 
-impl<'client> MqttClient {
+impl MqttClient {
     pub fn new(options: &MqttOptions) -> (Self, EventLoop) {
         let (client, event_loop) = AsyncClient::new(options.clone(), 1000);
         (MqttClient { client }, event_loop)

